c2corg_api/views/sitemap_xml.py

```diff
@@ -0,0 +1,222 @@
+from c2corg_api.caching import cache_sitemap_xml
+from c2corg_api.models.cache_version import CacheVersion
+from c2corg_api.models.document import Document, DocumentLocale
+from c2corg_api.models.route import ROUTE_TYPE, RouteLocale
+from c2corg_api.models.user_profile import USERPROFILE_TYPE
+from c2corg_api.views import cors_policy, etag_cache
+from c2corg_api.views.validation import create_int_validator, \
+    validate_document_type
+from c2corg_common.utils.caching import get_or_create
+from cornice.resource import resource, view
+from pyramid.httpexceptions import HTTPNotFound
+from sqlalchemy.sql.functions import func
+from math import ceil
+from datetime import date, datetime, timezone
+from slugify import slugify
+
+log = logging.getLogger(__name__)
+
+# Search engines accept not more than 50000 urls per sitemap,
+# and the sitemap files may not exceed 10 MB. With 50000 urls the sitemaps
+# are not bigger than 9MB, but to be safe we are using 45000 urls per sitemap.
+# see http://www.sitemaps.org/protocol.html
+PAGES_PER_SITEMAP = 45000
+
+
+UI_ENTRY_POINTS = {
+    'a': 'areas',
+    'b': 'books',
+    'c': 'articles',
+    'i': 'images',
+    'm': 'maps',
+    'o': 'outings',
+    'r': 'routes',
+    'w': 'waypoints',
+    'x': 'xreports'
+}
+
+validate_page = create_int_validator('i')
+
+
+@resource(
+    collection_path='/sitemaps.xml', path='/sitemaps.xml/{doc_type}/{i}.xml',
+    cors_policy=cors_policy, renderer='string')
+class SitemapXml(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @view()
+    def collection_get(self):
+        """ Returns a sitemap index file.
+        See: http://www.sitemaps.org/protocol.html
+
+        The response consists of a list of URLs of sitemaps.
+
+        <?xml version="1.0" encoding="UTF-8"?>
+            <sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+                <sitemap>
+                    <loc>https://api.camptocamp.org/sitemaps.xml/w/0.xml</loc>
+                    <lastmod>2019-02-11T18:01:49.193770+00:00</lastmod>
+                </sitemap>
+                <sitemap>
+                    <loc>https://api.camptocamp.org/sitemaps.xml/a/0.xml</loc>
+                    <lastmod>2019-02-11T18:01:49.193770+00:00</lastmod>
+                </sitemap>
+                <sitemap>
+                    <loc>https://api.camptocamp.org/sitemaps.xml/i/0.xml</loc>
+                    <lastmod>2019-02-11T18:01:49.193770+00:00</lastmod>
+                </sitemap>
+                <sitemap>
+                    <loc>https://api.camptocamp.org/sitemaps.xml/i/1.xml</loc>
+                    <lastmod>2019-02-11T18:01:49.193770+00:00</lastmod>
+                </sitemap>
+            </sitemap>
+        """
+        cache_key = _get_cache_key()
+        etag_cache(self.request, cache_key)
+
+        self.request.response.content_type = "text/xml"
+
+        return get_or_create(
+            cache_sitemap_xml, cache_key, _get_sitemap_index)
```

Comment:
no line return (does not seem needed?)

Otherwise something like 
```
return get_or_create(
            cache_sitemap_xml,
            cache_key,
            _get_sitemap_index
)
```

would be more readable IMHO